scenarios/csharp/dotnet/captioning/Program.cs

```diff
@@ -0,0 +1,423 @@
+            {
+                speechConfig.SetProperty(PropertyId.SpeechServiceResponse_StablePartialResultThreshold, stablePartialResultThresholdValue);
+            }
+            
+            if (userConfig.trueTextEnabled)
+            {
+                speechConfig.SetProperty(PropertyId.SpeechServiceResponse_PostProcessingOption, "TrueText");
+            }
+            
+            return speechConfig;
+        }
+
+        static SpeechRecognizer SpeechRecognizerFromSpeechConfig(SpeechConfig speechConfig, AudioConfig audioConfig, UserConfig userConfig)
+        {
+            SpeechRecognizer speechRecognizer;
+            if (userConfig.languageIDLanguages is string[] languageIDLanguagesValue)
+            {
+                AutoDetectSourceLanguageConfig detectLanguageConfig = AutoDetectSourceLanguageConfig.FromLanguages(languageIDLanguagesValue);
+                speechRecognizer = new SpeechRecognizer(speechConfig, detectLanguageConfig, audioConfig);
+            }
+            else
+            {
+                speechRecognizer = new SpeechRecognizer(speechConfig, audioConfig);
+            }
+            
+            if (userConfig.phraseList is string phraseListValue)
+            {
+                PhraseListGrammar grammar = PhraseListGrammar.FromRecognizer(speechRecognizer);
+                grammar.AddPhrase(phraseListValue);
+            }
+            
+            return speechRecognizer;
+        }
+
+        static SpeechRecognizer SpeechRecognizerFromUserConfig(UserConfig userConfig)
+        {
+            AudioConfig audioConfig = AudioConfigFromUserConfig(userConfig);
+            SpeechConfig speechConfig = SpeechConfigFromUserConfig(userConfig);
+            return SpeechRecognizerFromSpeechConfig(speechConfig, audioConfig, userConfig);
+        }
+
+        static async Task<string?> RecognizeContinuous(SpeechRecognizer speechRecognizer, UserConfig userConfig)
+        {
+            var recognitionEnd = new TaskCompletionSource<string?>();
+            int sequenceNumber = 0;
+
+            if (userConfig.partialResultsEnabled)
+            {
+                speechRecognizer.Recognizing += (object? sender, SpeechRecognitionEventArgs e) =>
+                    {
+                        if (ResultReason.RecognizingSpeech == e.Result.Reason && e.Result.Text.Length > 0)
+                        {
+// We don't show sequence numbers for partial results.
+                            WriteToConsoleOrFile(CaptionFromSpeechRecognitionResult(0, e.Result, userConfig), userConfig);
+                        }
+                        else if (ResultReason.NoMatch == e.Result.Reason)
+                        {
+                            WriteToConsole("NOMATCH: Speech could not be recognized.\n", userConfig);
+                        }
+                    };
+            }
+            else
+            {
+                speechRecognizer.Recognized += (object? sender, SpeechRecognitionEventArgs e) =>
```

Comment:
This might or might not be in scope, but if you want detailed recognition results with word-level offset and duration, set the following:
```csharp
// Get Words with Offset and Duration.
speechConfig.RequestWordLevelTimestamps();
```

Then we could show something like this: 
```csharp
speech_recognizer.Recognized += (object sender, SpeechRecognitionEventArgs e) =>
    {
        if (ResultReason.RecognizedSpeech == e.Result.Reason && e.Result.Text.Length > 0)
        {            
            Console.WriteLine($"RECOGNIZED: Text={e.Result.Text}");
            Console.WriteLine(String.Format ("Offset in Ticks: {0}", e.Result.OffsetInTicks));
            Console.WriteLine(String.Format ("Duration in Ticks: {0}", e.Result.Duration.Ticks));
                        
            var detailedResults = speechRecognitionResult.Best();
            if(detailedResults != null && detailedResults.Any())
            {
                // The first item in detailedResults corresponds to the recognized text.
                // This is not necessarily the item with the highest confidence number.
                var bestResults = detailedResults?.ToList()[0];
                Console.WriteLine($"\tConfidence: {bestResults.Confidence}\n\tText: {bestResults.Text}\n\tLexicalForm: {bestResults.LexicalForm}\n\tNormalizedForm: {bestResults.NormalizedForm}\n\tMaskedNormalizedForm: {bestResults.MaskedNormalizedForm}");
                
                // You must set speechConfig.RequestWordLevelTimestamps() to get word-level timestamps.
                Console.WriteLine($"\tWord-level timing:");
                Console.WriteLine($"\t\tWord | Offset | Duration");
                Console.WriteLine($"\t\t----- | ----- | ----- ");

                foreach (var word in bestResults.Words)
                {
                    Console.WriteLine($"\t\t{word.Word} | {word.Offset} | {word.Duration}");
                }
            }
        }
    };
```

Reply:
I don't know whether they want word-level timestamping. I'll leave this open until we know. This sounds like a good thing to bring up in today's meeting or in that e-mail thread if Katerina responds.